zerver/actions/realm_settings.py

```diff
@@ -691,6 +709,17 @@ def do_scrub_realm(realm: Realm, *, acting_user: UserProfile | None) -> None:
         acting_user=acting_user,
         event_type=AuditLogEventType.REALM_SCRUBBED,
     )
+    realm.scheduled_deletion_date = None
+    realm.save()
+
+
+def scrub_deactivated_realm(realm_to_scrub: Realm) -> None:
+    if (
+        realm_to_scrub.scheduled_deletion_date is not None
+        and realm_to_scrub.scheduled_deletion_date <= timezone_now()
+    ):
+        do_scrub_realm(realm_to_scrub, acting_user=None)
+        logging.info("Scrubbed realm %s", realm_to_scrub.id)
```

Comment:
It might make sense to assert that the realm is deactivated here; it'd be a bug for this to be called if it isn't.
(In general, when doing something destructive, we always want to be a little paranoid).

Reply:
Just noting here that the suggested assert was added in the merged changes.